File: Java/Easy/Part002/Lowest Mileage Car.java
import java.util.Scanner;

public class Hello {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine(); // Read the entire line of input

        String[] carInfos = input.split("\\s+");

        String lowestMileageCar = "";
        double minMileage = Double.MAX_VALUE; // Start with a very high mileage

        for (String carInfo : carInfos) {
            // Extract CARNAME and MILEAGE
            String carName = carInfo.substring(0, carInfo.indexOf('@'));
            double mileage = Double.parseDouble(carInfo.substring(carInfo.indexOf('@') + 1));

            // Check if this car has lower mileage than current minimum
            if (mileage < minMileage) {
                minMileage = mileage;
                lowestMileageCar = carName;
            }
        }

        System.out.println(lowestMileageCar); // Output the name of the car with lowest mileage
        scanner.close();
    }
}

File: python/Easy/Part002/LowestMileageCar.py
# Input
input_string = input()

# Splitting the input string by spaces to get each car's information
cars = input_string.split()

# Initialize variables to keep track of the car with the lowest mileage
lowest_mileage_car = None
lowest_mileage = float('inf')

# Iterate over each car's information
for car in cars:
    name, mileage = car.split('@')
    mileage = float(mileage)  # Convert mileage to float
    
    # Update the car with the lowest mileage
    if mileage < lowest_mileage:
        lowest_mileage_car = name
        lowest_mileage = mileage

# Output the car with the lowest mileage
print(lowest_mileage_car)
